### PR TITLE
Implement live network deploy script

### PR DESCRIPTION
**Pull request by pheuberger**
Also adds support to impersonate accounts via private key or mnemonic.

### CODE REVIEW

File: scripts/deploy-livenet.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const assert = require("assert");
```
Comment (kargakis) [inline scripts/deploy-livenet.js:1]: Is there any reason we cannot merge this and the `toucan-bridge.js` script?

Comment (pheuberger) [reply scripts/deploy-livenet.js:1]: holy moly, i completely missed that! the name was so misleading that i didn't even had a look inside 🙈  @dave are  you okay with merging the two?

Comment (kargakis) [reply scripts/deploy-livenet.js:1]: Btw we could even merge with deploy-local and just have an extra flag `--local=true` that sets up the local environment before deploying the bridge and could also enforce that the chain id is a local one. The benefit of this being that our deploy code is tested at all times via CI.

Comment (pheuberger) [reply scripts/deploy-livenet.js:1]: sounds good. i'll amend this PR 👍

**Comment by clevinson** [comment]
Adding @technicallyty as a reviewer from our side.

File: scripts/deploy-livenet.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const { deployBridge } = require("../lib/bridge");
+
+const { NCT_ADDRESS, BRIDGE_CONTROLLER_ADDRESS } = process.env;
+const zeroAddress = "0x0000000000000000000000000000000000000000";
+
+async function deploy() {
+	assert(NCT_ADDRESS, "NCT address needs to be set");
+	const tokenIssuer = BRIDGE_CONTROLLER_ADDRESS || zeroAddress;
```
Comment (technicallyty) [inline scripts/deploy-livenet.js:10]: should we also assert bridge controller address?  or is having the zero addr sometimes wanted?

**Review by technicallyty — APPROVED** [review]
lgtm, 1 question

File: scripts/deploy-livenet.js (hunk 1 above)
Comment (dave) [reply scripts/deploy-livenet.js:1]: Hey sorry I totally missed this notification last week... Yes I'm fine with rearranging the JS, but ping me when the PR is ready and I'll test it with the docker testing infrastructure we have in https://github.com/rnd-inc/toucan-bridge-service ... essentially as long as deploy-local.js can be coaxed into doing what it does now, I'm happy!

Comment (dave) [reply scripts/deploy-livenet.js:1]: BTW deploy-local.js now no longer calls `prepareToucanEnv`, it just calls `deployFixedContracts`... so it doesn't create the initial data, it just deploys the contracts. (The tests create their own initial data now).

**Comment by ryanchristo** [comment]
@pheuberger let us know if you want @technicallyty or myself to make the changes (i.e. merge the scripts)

**Comment by pheuberger** [comment]
> @pheuberger let us know if you want @technicallyty or myself to make the changes (i.e. merge the scripts)

hey @ryanchristo sorry i didn't get around to doing it, but yes that would be fabulous! :pray:

**Comment by ryanchristo** [comment]
@aspiers is this something we need or will this become irrelevant when we merge the contracts into the toucan monorepo?

**Comment by aspiers** [comment]
I believe we would need some bits of it but probably not the private key / mnemonic stuff.  @pheuberger please can you comment?

**Comment by pheuberger** [comment]
> I believe we would need some bits of it but probably not the private key / mnemonic stuff. @pheuberger please can you comment?

yeah I think so too. we don't need the PK and mnemonic handling, but would probably use the logic to deploy the bridge.

**Pull request closed without merging**